### PR TITLE
Add the git-psykorebase command.

### PR DESCRIPTION
**Pull request by Natim**
Git psykorebase let your do a rebase without changing the history and with only one conflict handling commit.

See https://github.com/benoitbryon/psykorebase

### CODE REVIEW

**Comment by Natim** [comment]
Here is my first proposal. r? @hemanth @benoitbryon

**Comment by hemanth** [comment]
LGTM, //cc @spacewander

**Comment by spacewander** [comment]
What about adding documentation for it under `./man`?

File: bin/git-psykorebase [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    git rev-parse --abbrev-ref HEAD
+}
+
+function usage()
+{
+    echo "USAGE: ${PROGRAM} PRIMARY_BRANCH [SECONDARY_BRANCH]"
+    echo "USAGE: ${PROGRAM} --continue"
+    echo ""
+    echo "OPTIONS:"
+    echo "  --no-ff:			Force rebase commit."
```
Comment (nicolaiskogheim) [inline bin/git-psykorebase]: The usage message doesn't say where the `--no-ff`-option goes :)

File: bin/git-psykorebase [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+CONTINUE="no"
+
+function current_branch() {
+    git rev-parse --abbrev-ref HEAD
+}
+
+function usage()
+{
+    echo "USAGE: ${PROGRAM} PRIMARY_BRANCH [SECONDARY_BRANCH]"
+    echo "USAGE: ${PROGRAM} --continue"
```
Comment (nicolaiskogheim) [inline bin/git-psykorebase]: I would argue that people understand what  ${0} is, so you can drop the variable. But you make the call.

File: bin/git-psykorebase (hunk 1 above)
Comment (Natim) [inline bin/git-psykorebase]: Actually you can put it where you want but I will add it.

**Comment by Natim** [comment]
Updated with the documentation.

**Comment by hemanth** [comment]
Cool :+1:

**Comment by Natim** [comment]
This is great thanks. Can we release a new version of git-extras to get this public?

**Comment by hemanth** [comment]
Yes, after #407 is resolved.

**Comment by benoitbryon** [comment]
Great! Thanks a lot!

**Pull request merged**